8.0 Add module account_bank_statement_import_paypal

alexis-via opened the pull request:
This module is designed for the new bank statement system of the PR #6 

This module is certainly not perfect, in particular on the problems of multi-language. The problem is that the Paypal CSV files use strings in the language of the user... and I use a few of those strings in the parsing. But I don't see any easy solution for this...

Commit:
Add module account_bank_statement_import_paypal

Commit:
Add module account_bank_statement_import_paypal

pedrobaeza commented:
For the parsing question, I would use _() to also translate the string you are using. What do you think?

max3903 commented on account_bank_statement_import_paypal/account_bank_statement_import_paypal.py:
```diff
@@ -0,0 +1,201 @@
+        vals_line = False
+        user = self.pool['res.users'].browse(cr, uid, uid, context=context)
+        company_currency_name = user.company_id.currency_id.name
+        comission_total = 0.0
```
commission

Commit:
Fix spelling mistake

Commit:
Fix spelling mistake

Commit:
Add methods to inherit for language-specific or country-specific stuff
Port to new API

Commit:
Add methods to inherit for language-specific or country-specific stuff
Port to new API

alexis-via commented:
@pedrobaeza I think it's not as easy as that. For example: I use OpenERP in English because I prefer to use it in english to avoid all the bad French translation and to avoid the mix of translated and untranslated strings. But my Paypal bank statements are in French because Paypal knows I'm from France.

I think that the best solution is to have some _prepare method that can be inherited in a country-specific module. It is needed for:
- date parsing
- file encoding
- amount parsing (decimal separator)
- status of the line

i just implemented it. Tell me what you think.

pedrobaeza commented:
You can add then a parameter to say in which language is the file, and use .with_context(lang='...') in all parse calls.

alexis-via commented:
@pedrobaeza I'm really not sure that using translations is the good answer for this. The problem is that I don't have enough information on Paypal CSV bank statements to decide what is the best solution, that's why I propose to use the _prepare_*() solution for the moment.

pedrobaeza commented:
But this is also not a good option, because this forces to make subsequent account_bank_statement_import_paypal_xx for each language, and with no possibility of mixing paypal files from one language and another. We should provide a configurable mechanism.

Commit:
FIX super

Commit:
FIX super

lmignon committed:
[FIX] bank_account created by the import must belong to the partner linked to the company of the provided journal
PR on master: odoo/odoo#5821

lmignon committed:
[IMP] Backport from master at 04daefd2d101d90daf5782173b95f31f3bd9e1b6

lmignon committed:
[IMP] Add sanitezed_acc_number on res.partenr.bank
The new field is computed by sanitzing the acc_number.

lmignon committed:
[IMP] Improve test coverage

lmignon committed:
[IMP] Sanitize acc_number to upper case
Use wildcard on both sides when opertor contains 'like'

Commit:
[RFR] Base import parsers on new api and master.

Commit:
[ENH] Support multiple accounts/currencies.

Copy of changes proposed for master.

Commit:
[RFR] Complete streamlining/refactoring of bank statement import.

bank_statement_parse module now contains only one library that can be used to convert 7.0 parsers,
or to provide a common reference for new parsers, providing some independence of the database structure for
bank statements in base Odoo.

Commit:
[FIX] Add back __unported__ to prevent deletions after PR.

Commit:
[ENH] Add enhanced mt940 import.

Commit:
[FIX] Update dependencies.

Commit:
[ENH] Support both Rabo and ING MT940.

Commit:
[FIX] Whitespace error.

Commit:
[FIX] Do not specify iban for bank in demo data.

Commit:
[FIX] But state is required for res.partner.bank.

Commit:
[FIX] No journal_id keyword parm on _create_bank_account(..)

lmignon committed:
[IMP] Let the res_partner_bank.post_write create the journal if it doesn't exist

lmignon committed:
[IMP] Specify the currency to use for the account.account and account.journal created by res_partner_bank.post_write

lmignon committed:
[FIX] backport from MP on upstream until 5f30c6ed09e92e53ad870d1fd54288ac43a4ca9e

sbidoul commented:
@alexis-via there is a pep8 issue:

`./account_bank_statement_import_paypal/account_bank_statement_import_paypal.py:164:25: W503 line break before binary operator`

And unicodecsv must be added to the travis config.

Commit:
[7.0] Forward port improvements in 7.0 branch.

Commit:
[FIX] No longer include __unported__ modules.

Commit:
[FIX] Use unique accounts and improved bank search in base.

Commit:
[FIX] No index=True needed when specifying unique constraint.

lmignon committed:
[IMP] use new API

Conflicts:
	account_bank_statement_import/account_bank_statement_import.py

hbrunn committed:
[ADD] migration script

and a module that takes care of saving imported files (+migration)

hbrunn committed:
[ADD] test

hbrunn committed:
[IMP] make test work with and without demo data

Commit:
Update travis config

Commit:
Update travis config

hbrunn committed:
[FIX] account.bank.statement.import is transient

lmignon committed:
[IMP] Extract description to README
Add OCA as author,
PEP8

lmignon committed:
[ADD] i18n

lmignon committed:
[FIX] Keep icon retouched by @pedrobaeza

hbrunn committed:
[FIX] unused imports

lmignon committed:
[IMP]Move the field journal_id in the main module.
By moving the field in the main module the others parsers can use it without duplicate it

pedrobaeza committed:
Merge pull request #19 from acsone/8.0-refactor-new-api

8.0 refactor new api

nbessi committed:
account_bank_statement_import_ofx is not automatically installed anymore

pedrobaeza committed:
Merge pull request #23 from nbessi/account_bank_statement_import_ofx_no_auto_install

account_bank_statement_import_ofx is not automatically installed anymore

Commit:
[ENH] Better separation of different MT940 types.

nbessi committed:
Avoid demo data using an exisiting account number

Fixes #26

Commit:
[FIX] Fixed some problems after rebasing.

Commit:
[FIX] Test of ofx should not fail due to USD currency used.

Commit:
[FIX] Company and statement currency can be different.

Commit:
[FIX] Now succesfully running tests for all imports.

Commit:
[FIX] Solving pylint/flake and test errors.

Also added dutch translation.

Commit:
[FIX] Attempt to stop flake8 from complaining.

Commit:
[ENH] Add back support for both old and new ING mt940 format.

Commit:
[ENH] Support zip files.

Commit:
[FIX] Updated headers, README.rst's and removed vim lines.

Commit:
[FIX] One line to long in mt940.py.

Commit:
[FIX] Copyright headers, vim line and manifest keys.

Commit:
[FIX] Prevent unneeded processing and memory use in parsers.

Commit:
[FIX] Restructure and expand tests. Fix some bugs.

There is now a common class to test import of statements and their transactions.
Fixes:
- Field in regular expression for mt940 rabo parser was to short for bank-account.
- Wrong name used for reference attribute in mt940 rabo parser;
- Attribute transferred_amount should not have been renamed amount in BankTransaction.

Commit:
[FIX] Removed unneeded (and malformed) comment line.

sbidoul committed:
Merge pull request #27 from nbessi/fix_demo_data

Avoid demo data using an exisiting account number

pedrobaeza committed:
[FIX] account_bank_statement_import: Handle correctly the selection of the journal

antespi committed:
[FIX] Change default by compute and between quotes in order to allow inheritance

pedrobaeza committed:
Merge pull request #1 from Antiun/8.0-accont_bank_statement_import-journal-fix

[FIX] Change default by compute and between quotes in order to allow …

pedrobaeza committed:
[FIX] account_bank_statement_import: PEP8

hbrunn committed:
[UPD] adapt to upstream changes

hbrunn committed:
[FIX] unused import

pedrobaeza committed:
Merge pull request #30 from pedrobaeza/8.0-account_bank_statement_import-journal

[FIX] account_bank_statement_import: Handle correctly the selection of the journal

kvdb commented:
I've been using this PR with a couple of minor adjustments for localization purposes and a rebase on PR #15. It usually works fine, although one of the CSV exports couldn't be parsed.

I encountered this error:

```
Value '...' for the field 'commission' on line 14, cannot be converted to float 
```

Line 14 (I replaced some text, but only the sensitive data. Nothing related to those triple dot values):

```
"06-07-2014","19:16:35","GMT+02:00","ThinkGeek, Inc.","Order","Completed","USD","-132,80","...","...","info@foo.bar","paypal@thinkgeek.com","O-28941779A1111111111","","Non-Confirmed","","B19A6,promotion","","","","","","","","","","","","","","","5b711e1ae","","","...","Torenallee 20","","Eindhoven","NOORD-BRABANT","5617 BC","Netherlands","",
```

Commit:
[FIX] _get_journal method has one less argument.

kvdb commented on account_bank_statement_import_paypal/README.rst:
```diff
@@ -0,0 +1,41 @@
+Import Paypal Bank Statements
+=============================
+
+This module allows you to import the Paypal CSV files in Odoo as bank statements.
```
You might want to describe which kind of PayPal CSV file exports are accepted. I can download 3 types (all activity, balance affecting payments or completed payments) from PayPal of which sometimes only the balance affecting payments imports properly.

kvdb commented:
Another issue I noted: one transaction is showing up on the reconciliation screen as 'To British Pound : 5AL64185U4882084C" instead of a reference to "thetrainline". These are the CSV lines that lead up to it:

```
"21-09-2014","13:07:35","GMT+02:00","thetrainline","Express Checkout Payment Sent","Completed","GBP","-20,15","0,00","-20,15","info@foo.bar","ttl_paypal@thetrainline.com","2QM72241XA3211636","Non-U.S. - Verified","","","","0,00","","0,00","","","","","","","","","","","02D153823V1084457","2114079950","","","0,00","","","","","","","",
"21-09-2014","13:07:35","GMT+02:00","From Euro","Currency Conversion","Completed","GBP","20,15","0,00","20,15","","","0MD85557EX105261C","","","","","","","","","","","","","","","","","","2QM72241XA3211636","","","","20,15","","","","","","","",
"21-09-2014","13:07:35","GMT+02:00","To British Pound","Currency Conversion","Completed","EUR","-26,58","0,00","-26,58","","","5AL64185U4882084C","","","","","","","","","","","","","","","","","","2QM72241XA3211636","","","","1.027,69","","","","","","","",
"21-09-2014","13:07:35","GMT+02:00","PayPal","Temporary Hold","Removed","EUR","26,58","0,00","26,58","","","96K33589W03715300","","","","","","","","","","","","","","","","","","7TV55533PV057114D","","","","1.054,27","","","","","","","",
"21-09-2014","13:07:32","GMT+02:00","PayPal","Temporary Hold","Placed","EUR","-26,58","0,00","-26,58","","","7TV55533PV057114D","","","","","","","","","","","","","","","","","","02D153823V1084457","","","","1.027,69","","","","","","","",
```

![train](https://cloud.githubusercontent.com/assets/97039/8377809/c81033d2-1c14-11e5-8527-e93dfa5a0150.png)

Commit:
[ENH] Support both old and new style parse results.

hbrunn committed:
Revert "[FIX] Use unique accounts and improved bank search in base."

This reverts commit 5538c234d9660224c6d37b0b1888798c22afc8c9.

hbrunn committed:
Revert "[FIX] No index=True needed when specifying unique constraint."

This reverts commit 1dcd07e8931f4519dcbd23535415d379b1db001f.

hbrunn committed:
[FIX] pass parameters we got to super

pedrobaeza committed:
Merge pull request #32 from hbrunn/8.0-fix_res_partner_bank_search

[FIX] pass parameters we got to super

sbidoul committed:
[IMP] add note field in parserlib

sbidoul committed:
[IMP] lower case _LOGGER and log some more exception details

Commit:
Merge pull request #5 from acsone/8.0-stmt-import-1-sbi

A few improvements to PR #15

Commit:
[FIX] Do not override statement_id in mt940.

If statement_id is filled with a proper value (not zeroes and/or whitespace), do not override
it in the mt940 parser.

sbidoul committed:
[IMP] rename bank_statement_parse_ to account_bank_statement_import_

Move parserlib to account_bank_statement_import and remove bank_statement_parse module

sbidoul committed:
[FIX] pep8

Commit:
Merge pull request #6 from acsone/8.0-rename-sbi

[IMP] rename bank_statement_parse_ to account_bank_statement_import_

sbidoul committed:
Merge pull request #15 from NL66278/8.0_import_parsers_new_api_therp

8.0 import parsers new api therp

gurneyalex committed:
Update travis configuration

use sudo: false to enable container build
use cache: pip to cache pip packages

hbrunn committed:
[ADD] configuration section on accounting page

[REN] move files around to conform directory convention

hbrunn committed:
[FIX] config settings are transient models

hbrunn committed:
[ADD] base_bank_account_number_unique

hbrunn committed:
[IMP] better icon

hbrunn committed:
[FIX] excape backslash

pedrobaeza committed:
Merge pull request #21 from hbrunn/8.0-migrate_from_7.0_therp

[ADD] migration script

pedrobaeza committed:
Merge pull request #31 from hbrunn/8.0-no_uniqueness_in_sanitized_account_number

8.0 no uniqueness in sanitized account number

Commit:
[FIX] Make sure old style parse results are handled properly.

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

sbidoul committed:
[ADD] addons table in README.md

lmignon committed:
The import must fail if the bank_account or the account_journal is not found
The creation of account_journal and bank_account must be prohibited when importing statments.

lmignon committed:
Bump version number

pedrobaeza committed:
Merge pull request #37 from acsone/8.0-fix-prevent-create-bank_anccount-and-account_journal-lmi

The import must fail if the bank_account or the account_journal is no…

Commit:
[FIX] Import exceptions.Warning as UserError.

This according to OCA guidelines.
Also reformat README.rst to keep lines within limit.

sbidoul committed:
[UPD] addons table in README.md

rschnapka committed:
german translations for bank-statement-import

oca-transbot committed:
OCA Transbot updated translations from Transifex

sbidoul committed:
[UPD] prefix versions with 8.0

Commit:
[ENH] Make sure message is filled for camt.

Message was not taken from AddtlNtryInf element, if not part of details.

Not sure wether it actually can be part of details (NtryDtls element), but this change least intrusive.

oca-transbot committed:
OCA Transbot updated translations from Transifex

StefanRijnhart committed:
Merge pull request #39 from NL66278/8.0_camt_name

[ENH] Make sure message is filled for camt.

pedrobaeza committed:
Merge pull request #38 from rschnapka/8.0

german translations for bank-statement-import

Commit:
[FIX] Some finetuning of version numbers and #pylint comments.

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

alexis-via committed:
FIX 2 variables had the same name (account_number) !!!

alexis-via committed:
PEP8 fix

StefanRijnhart committed:
Merge pull request #42 from akretion/8.0-fix-bug-wrong-variable-name

8.0 FIX 2 variables had the same name (account_number)

oca-transbot committed:
OCA Transbot updated translations from Transifex

hbrunn committed:
[FIX] don't error out if there are no attachments

pedrobaeza committed:
Merge pull request #43 from hbrunn/8.0-account_bank_statement_import_save_file-init_hook_no_attachments

[FIX] don't error out if there are no attachments

hbrunn committed:
[FIX] have one list of notifications

[IMP] use a nice formatting to show notifications for imported file

pedrobaeza committed:
Merge pull request #47 from hbrunn/8.0-notifications_one_list

[FIX] have one list of notifications

hbrunn committed:
Merge pull request #36 from NL66278/8.0_import_parsers_complete_therp

[ADD] support zip files

hbrunn committed:
[FIX] incomplete format string

sbidoul committed:
Merge pull request #50 from hbrunn/8.0-account_bank_statement_import_save_file-incomplete_format

[FIX] incomplete format string

OCA-git-bot committed:
[UPD] addons table in README.md

zecke commented:
Changes to import German PayPal CSV:
- Match for 'Abgeschlossen' for complete statements.
- Check for 'Datum' instead of Date
- Numbers are 3.100,55.. and need to change
- balance is column 22 (not 34), transac_ref is column 18 (not 30).

oca-transbot committed:
OCA Transbot updated translations from Transifex

Commit:
[ENH] Might need info in camt import from PmtInfId element.

Commit:
[FIX][account_bank_statement_import_paypal] Consider refund line

Commit:
[FIX][account_bank_statement_import_paypal] Consider refund line

Commit:
[FIX][account_bank_statement_import_paypal] Avoid to return paypal_email_account as a bank account

Commit:
[FIX][account_bank_statement_import_paypal] Avoid to return paypal_email_account as a bank account

Commit:
[IMP][account_bank_statement_import_paypal] Put file transaction ref in name to work with base_transaction_ref

Commit:
[IMP][account_bank_statement_import_paypal] Put file transaction ref in name to work with base_transaction_ref

Commit:
[FIX][account_bank_statement_import_paypal] Remove thousands separator

Commit:
[FIX][account_bank_statement_import_paypal] Remove thousands separator

Commit:
[FIX] Correct indentation in camt.py.

alexis-via committed:
Add support for OFX files without <NAME> inside all <STMTTRN>

Add a comment in the code to explain it all !

adrienpeiffer commented:
Thank you for your work!

I tested this with several files. The format seems to be the same as you.

But I encounter some problems:
- Refund lines are ignored, or I think we need to import them.
- To work properly with base_transaction_ref module the id of the transaction should be in the name field of the bank statement line.
- Paypal_email_account is returned in _parse_file is treated as a bank account if that raises an exception.
- The thousand separator is not considered in the transformation in a float number.

I'll do a PR about it on your branch.

Thanks,
Adrien

oca-transbot committed:
OCA Transbot updated translations from Transifex

alexis-via committed:
Better fix: fallback on <TRNTYPE> (required in OFX) and also <CHECKNUM> if we have it

alexis-via committed:
FIX when transaction.payee is not set: don't detect wrong bank_account/partner (it used to match that res.partner.bank that have an empty 'owner_name')

alexis-via committed:
Better error message

alexis-via commented:
Just to make things clear:
1) I don't think OCA will ever merge this PR ; it there were some motivation of the reviewers to merge it, it would have been merged long ago.
2) I don't plan to work on this PR myself anymore.
3) these Paypal bank statement are a piece of !@*%$£§&#, without strict formatting, with a very strange behavior for transactions in other currencies, with stupid handling of languages and localisations, etc... really something that Paypal cannot be proud of. I really hope that Paypal will fix it one day and offer a better format for their bank statement... but I don't know if it will happen one day.

So feel free to take over this PR. If you think that the reviewers could maybe merge it one day, you can make some PR in this PR and I'll merge them immediately.

Commit:
Merge pull request #2 from acsone/8.0-paypal-acsone-ape

[8.0] Paypal

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

hbrunn committed:
Merge pull request #55 from akretion/8.0-better-error-msg

8.0 Better error messages

hbrunn committed:
Merge pull request #54 from akretion/8-ofx-no-name-tag

8.0 Add support for OFX files without <NAME> inside all <STMTTRN>

hbrunn committed:
[IMP] if there's a date but no period, ger the date's period

hbrunn committed:
[IMP] pass account_period_prefer_normal

pedrobaeza committed:
Merge pull request #58 from hbrunn/8.0-set_period_id_from_date

[IMP] if there's a date but no period, ger the date's period

Commit:
OFX: Stronger unique_import_id (to lower the risk of having 2 identical unique_import_id in the same file)

Use bank account number as bank statement label, instead of bank number (important when you have several bank accounts at the same bank)

alexis-via committed:
OFX: Display both bank number and account number in bank statement name

Adapt automated test

legalsylvain committed:
Merge pull request #51 from NL66278/8.0_camt_name

[ENH] Might need info in camt import from PmtInfId element.

oca-transbot committed:
OCA Transbot updated translations from Transifex

OCA-git-bot committed:
[ADD] setup.py

alexis-via committed:
Shorter bank statement name, as requested by Pedro

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
Merge pull request #62 from akretion/8.0-ofx-stronger-unique_import_id

OFX: stronger unique_import_id

mourad-ehm committed:
Merge branch '8.0-paypal' of https://github.com/akretion/bank-statement-import into 8.0-paypal

mourad-ehm committed:
[FIX] adapt code to the new paypal format

mourad-ehm committed:
[FIX] error when several line defined in currency != company currency

pedrobaeza commented:
As there's no plans to work on this from the author, I close. Feel free to reopen in case there's interest.

elicoidal commented:
@pedrobaeza @jbeficent Actually this coould be interesting to fix for the particular case of OCA Paypal Bank Statement.

Would it be worth the effort to fix it only for our particular case?

pedrobaeza commented:
Well, it's very specific to the downloaded format, but if you pass me a sample, I can check the coincidences. I have set up a variant of this one for an Spanish Paypal statement.

elicoidal commented:
Thanks I will private email you
